sunpycube/cube/NDCube.py

```diff
@@ -0,0 +1,385 @@
+
+    meta : dict-like object, optional
+        Additional meta information about the dataset. If no meta is provided
+        an empty collections.OrderedDict is created. Default is None.
+
+    unit : Unit-like or str, optional
+        Unit for the dataset. Strings that can be converted to a Unit are allowed.
+        Default is None.
+
+    copy : bool, optional
+        Indicates whether to save the arguments as copy. True copies every attribute
+        before saving it while False tries to save every parameter as reference.
+        Note however that it is not always possible to save the input as reference. Default is False.
+    """
+
+    def __init__(self, data, uncertainty=None, mask=None, wcs=None, meta=None, unit=None, copy=False, missing_axis=None, **kwargs):
+        axtypes = list(wcs.wcs.ctype)
+        array_order = sunpycube.cube.cube_utils.select_order(axtypes)
+        result_data = data.transpose(array_order)
+        wcs_order = np.array(array_order)[::-1]
+        result_wcs = sunpycube.wcs_util.reindex_wcs(wcs, wcs_order)
+        super(NDCubeOrdered, self).__init__(result_data, uncertainty=uncertainty, mask=mask,
+                                            wcs=result_wcs, meta=meta, unit=unit, copy=copy, missing_axis=missing_axis, **kwargs)
+
+
+def _plot_3D_cube(cube, image_axes=None, unit_x_axis=None, unit_y_axis=None,
+                  axis_ranges=None, **kwargs):
+    """
+    Plots an interactive visualization of this cube using sliders to move through axes
+    plot using in the image.
+    Parameters other than data and wcs are passed to ImageAnimatorWCS, which in turn
+    passes them to imshow.
+
+    Parameters
+    ----------
+    image_axes: `list`
+        The two axes that make the image.
+        Like [-1,-2] this implies cube instance -1 dimension
+        will be x-axis and -2 dimension will be y-axis.
+
+    unit_x_axis: `astropy.units.Unit`
+        The unit of x axis.
+
+    unit_y_axis: `astropy.units.Unit`
+        The unit of y axis.
+
+    axis_ranges: `list` of physical coordinates for array or None
+        If None array indices will be used for all axes.
+        If a list it should contain one element for each axis of the numpy array.
+        For the image axes a [min, max] pair should be specified which will be
+        passed to :func:`matplotlib.pyplot.imshow` as extent.
+        For the slider axes a [min, max] pair can be specified or an array the
+        same length as the axis which will provide all values for that slider.
+        If None is specified for an axis then the array indices will be used
+        for that axis.
+    """
+    i = ImageAnimatorWCS(cube.data, wcs=cube.wcs, unit_x_axis=unit_x_axis, unit_y_axis=unit_y_axis,
+                         axis_ranges=axis_ranges, **kwargs)
+    return i
+
+
+def _plot_2D_cube(cube, axes=None, image_axes=['x', 'y'], **kwargs):
+    """
+    Plots an x-y graph at a certain specified wavelength onto the current
```

Comment:
Replace ```x-y graph at a certain specified wavelength``` with ```2D image```.  NDCube does not require a wavelength axis.  And ```x-y graph``` implies a 1D plot.